var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -576,6 +545,8 @@ def cmake_args(self):
             define("LLVM_REQUIRES_RTTI", True),
             define("LLVM_ENABLE_RTTI", True),
             define("LLVM_ENABLE_EH", True),
+            define("LLVM_ENABLE_TERMINFO", False),
```

Comment:
If package `ncurses` is needed (such as `+lldb`), should `LLVM_ENABLE_TERMINFO` be True?

In the case `ncurses` is needed, if `LLVM_ENABLE_TERMINFO` is False, then some symbols may be undefined.

Reply:
Indeed, this is failing to build currently due to undefined symbols from -ltinfo, as it's linking `-lcurses libform.so libpanel.so` but not `-ltinfo` (this also picks up the wrong libcurses.so from system rather than `-lncurses` from spack, but that's a separate issue):
```
/mnt/sw/nix/store/3w12p7w4kc0pnig6h14r1g5d9s9scbij-binutils-2.38/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `curses::Menu::DrawMenuTitle(curses::Window&, bool)':
IOHandlerCursesGUI.cpp:(.text._ZN6curses4Menu13DrawMenuTitleERNS_6WindowEb+0x17a): undefined reference to `acs_map'
```

Reply:
> Indeed, this is failing to build currently due to undefined symbols from -ltinfo, as it's linking `-lcurses libform.so libpanel.so` but not `-ltinfo` (this also picks up the wrong libcurses.so from system rather than `-lncurses` from spack, but that's a separate issue):
> 
> ```
> /mnt/sw/nix/store/3w12p7w4kc0pnig6h14r1g5d9s9scbij-binutils-2.38/bin/ld: lib/liblldbCore.a(IOHandlerCursesGUI.cpp.o): in function `curses::Menu::DrawMenuTitle(curses::Window&, bool)':
> IOHandlerCursesGUI.cpp:(.text._ZN6curses4Menu13DrawMenuTitleERNS_6WindowEb+0x17a): undefined reference to `acs_map'
> ```

This is because `ncurses` package doesn't create a symlink libcurses -> libncurses. I have a PR here that fixes this and in so doing resolves the build error:
* https://github.com/spack/spack/pull/32037